HCSP.js

```diff
@@ -0,0 +1,231 @@
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// Usage of Zotero.Utilities.xpath() function. Exist only since Zotero 2.1
+// https://www.zotero.org/support/dev/client_coding/changes_in_zotero_2.1#xpath_utility_functions
+// Then minimal version of Zotero needed is 2.1
+
+// TODO: Use camelStyle for variables to be consistent with Zotero coding style
+
+function detectWeb(doc, url) {
+    /* The URLs have upper or lower case on the website
+    depending the way you access to a page */
+    if (url.indexOf("avisrapportsdomaine") != -1) {
+        return 'report';
+    } else if (url.indexOf("avisrapports") != -1) {
+        return 'multiple';
+    }
+}
+
+function doWeb(doc, url) {
+
+    if (detectWeb(doc, url) == "multiple") {
+
+        var allItems = getSearchResults(doc);
+
+        Zotero.selectItems(allItems, function(items) {
+            if (!items) return true;
+
+            var articlesURLs = [];
+            for (var itemURL in items) {
+                articlesURLs.push(itemURL);
+                }
+            ZU.processDocuments(articlesURLs, scrape);
+        })
+
+    } else {
+        scrape(doc, url);
+    }
+}
+
+// For multiple articles
+function getSearchResults(doc) {
+    // Flag to check if something was found
+    var found = false;
+
+    // Titles and URLs of each item are on the third column of the only table
+    // This column is the only one with links (<a>)
+    var xpath_titles = '//tbody/tr/td/a';
+    var titles = ZU.xpath(doc, xpath_titles);
+
+    // Put all the titles and their URLs in the items object
+    var allItems = {};
+
+    for (var i = 0; i< titles.length; i++) {
+        var nextTitle = titles[i];
+        allItems[nextTitle.href] = nextTitle.textContent;
+        found = true;
+    }
+
+    // if no item found (no result in search), return false, else the items
+    return found ? allItems : false;
+}
+
+function scrape(doc, url) {
+
+    // init a new Zotero's item
+    var newItem = new Zotero.Item("report");
+
+    /*---------- Constants items ----------*/
+    newItem.publicationTitle = "Rapport de l'HCSP";
+    newItem.language = "fr-FR";
+    newItem.institution = "Haut Conseil de la Santé Publique";
+    newItem.place = "Paris";
+
+    // Creator is the institution (HCSP)
+    // Names of the author are inside the PDFs (if there is one)
+    newItem.creators.push({"lastName": "HCSP", "creatorType": "author", "fieldMode": 1});
+
+    /*---------- URL ----------*/
+    newItem.url = url;
+    
+    /*---------- Title ----------*/
+    var xpathTitle = '//li[@class="active"]';
+    var title = ZU.xpathText(doc, xpathTitle);
+    newItem.title = title;
+ 
+    /*---------- Date ----------*/
+    // Date is not easy to find because it is not in a named bloc. It is
+    // somewhere in one of the <p> in the div 'doc-donnees'
+    // Then to find it, use a Regex on each <p> bloc
```

Comment:
Since you're not doing anything complicated with your regexp, you can actually incorporate it into your xpath:
```
var date = ZU.xpath(doc, "//div[@id='doc-donnees']/p[contains(text(),'Date du document')][0]");
if (date) {
  date = date.textContent.match(/\b(\d{2})\/(\d{2})\/(\d{4})\b/); // note the \b
  if (date) {
    newItem.date = date[3] + "-" + date[2] + "-" + date[1];
  }
}